test_socket.py

```diff
@@ -0,0 +1,32 @@
+import server
+import pytest
+import socket
+import string
+import pdb
+import datetime
+
+@pytest.fixture(scope='session')
+def connection():
+    ADDR = ('127.0.0.1', 7070)
+    client = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+    )
+    client.connect(ADDR)
+    return client
+
+def test_client(connection):
+    connection.sendall('abc')
+    connection.shutdown(socket.SHUT_WR)
+    part = ""
+    while True:
+        tempPart = connection.recv(16)
+        part = part + tempPart
+        if len(tempPart) < 16:
+            connection.close()
+            break
+    part_list = string.split(part, '\r\n')
+    assert part_list[0].find("200 OK") != -1
+    date = datetime.datetime.strftime(datetime.datetime.now(), '%Y-%m-%d %H:%M')
```

Comment:
there's a non-trivial chance that this test will fail even though the response is technically correct.  You are comparing the current time when the test runs with the current time when the response is built.  That might not match, even on a minute border.